src/VirtualClient/VirtualClient.Contracts/Extensions/MetricExtensions.cs

```diff
@@ -0,0 +1,188 @@
+                            filteredMetrics = filteredMetrics.Where(m => m.Verbosity == MetricVerbosity.Standard);
+                            break;
+                        case "metricverbosity:informational":
+                            filteredMetrics = filteredMetrics.Where(m => m.Verbosity == MetricVerbosity.Informational);
+                            break;
+
+                        default:
+                            filteredMetrics = Enumerable.Empty<Metric>();
+                            break;
+                    }
+
+                    filterTerms = filterTerms.Where(f => !f.Contains(nameof(MetricVerbosity)));
+                }
+
+                if (filterTerms?.Any() == true)
+                {
+                    filteredMetrics = metrics.Where(m => filterTerms.Contains(m.Name, MetricFilterComparer.Instance)).ToList();
+                }
+            }
+
+            return filteredMetrics;
+        }
+
+        /// <summary>
+        /// Print metrics to console. Default to critical only.
+        /// </summary>
+        /// <param name="metrics">List of Metrics</param>
+        /// <param name="scenario">Scenario</param>
+        /// <param name="criticalOnly">Boolean to note if prints critical only metric</param>
+        public static void PrintConsole(this IEnumerable<Metric> metrics, string scenario, bool criticalOnly = true)
+        {
+            try
+            {
+                Console.WriteLine();
+                Dictionary<string, int> colWidths = new Dictionary<string, int>();
+
+                DataTable table = new DataTable();
+                table.Columns.Add("Scenario", typeof(string));
+                table.Columns.Add("Name", typeof(string));
+                table.Columns.Add("Value", typeof(double));
+                table.Columns.Add("Unit", typeof(string));
+
+                IEnumerable<Metric> metricsToPrint = criticalOnly ? metrics.Where(m => m.Verbosity == MetricVerbosity.Critical).ToList() : metrics;
+
+                foreach (Metric metric in metricsToPrint)
+                {
+                    DataRow row = table.NewRow();
+                    row["Scenario"] = scenario; // You can modify this as needed
+                    row["Name"] = metric.Name;
+                    row["Value"] = metric.Value;
+                    row["Unit"] = metric.Unit;
+                    table.Rows.Add(row);
+                }
+
+                MetricExtensions.PrintDataTableFormatted(table);
+            }
+            catch (Exception exc)
+            {
+                // Don't throw on print console errors
+                Console.WriteLine($"Failed to print metrics on console: '{exc.Message}'");
+            }
+        }
+
+        /// <summary>
+        /// Data table with formated column width.
+        /// </summary>
+        /// <param name="dataTable">Input data table.</param>
+        private static void PrintDataTableFormatted(DataTable dataTable)
```

Comment:
Naming: LogConsole

We don't need to be so explicit in the names of methods that are extensions and that have different input parameters. It adds unnecessary layers of abstraction.